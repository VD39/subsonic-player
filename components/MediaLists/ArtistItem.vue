<script setup lang="ts">
import ImageLink from '@/components/Image/ImageLink.vue';

const props = defineProps<{
  artist: Artist;
}>();

const toLink = `/artist/${props.artist.id}`;
</script>

<template>
  <article>
    <ImageLink :to="toLink" :title="artist.name" :image="artist.image" />

    <p class="clamp clamp2">
      <NuxtLink :to="toLink" :title="`Go to artist ${artist.name}`">
        {{ artist.name }}
      </NuxtLink>
    </p>
  </article>
</template>
