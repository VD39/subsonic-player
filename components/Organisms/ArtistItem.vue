<script setup lang="ts">
import ImageLink from '@/components/Organisms/ImageLink.vue';

const props = defineProps<{
  artist: Artist | SimilarArtist;
}>();

const toLink = `/artist/${props.artist.id}`;
</script>

<template>
  <article>
    <ImageLink
      :to="toLink"
      :title="`Go to artist ${artist.name}`"
      :image="artist.image"
      class="mBS"
    />

    <p class="mBXS strong smallFont clamp2">
      <NuxtLink
        :to="toLink"
        class="link"
        :aria-label="`Go to artist ${artist.name}`"
      >
        {{ artist.name }}
      </NuxtLink>
    </p>
  </article>
</template>
