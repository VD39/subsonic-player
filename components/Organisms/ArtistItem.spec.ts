import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import { formattedArtistMock } from '@/test/fixtures';

import ArtistItem from './ArtistItem.vue';

function factory(props = {}) {
  return mount(ArtistItem, {
    props: {
      artist: formattedArtistMock,
      ...props,
    },
  });
}

describe('ArtistItem', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
