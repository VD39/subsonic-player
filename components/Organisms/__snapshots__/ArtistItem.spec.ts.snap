// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArtistItem > matches the snapshot 1`] = `
"<article>
  <figure class="mBS"><a draggable="false" title="Go to artist name">
      <div class="centerAll preloadImage"><span class="skeletonLoader"><span class="visuallyHidden">Loading image</span></span><img alt="Go to artist name" class="image" draggable="false" src="image" style="display: none;"></div>
    </a></figure>
  <p class="mBXS strong smallFont clamp2"><a aria-label="Go to artist name" class="link" draggable="false">name</a></p>
</article>"
`;
